### PR TITLE
bpo-41115: Modified src to raise  rather `Unicode{Encode, Decode}Error` rather than plain `UnicodeError`

### PR DESCRIPTION
**Pull request by utkarsh261**
https://bugs.python.org/issue41115

### CODE REVIEW

File: Lib/encodings/idna.py [hunk 1]
```diff
@@ -79,17 +80,18 @@ def ToASCII(label):
...
-    except UnicodeError:
+    except UnicodeEncodeError:
         pass
     else:
         # Skip to step 8.
         if 0 < len(label) < 64:
             return label
-        raise UnicodeError("label empty or too long")
+        raise UnicodeEncodeError("ascii", label.decode("ascii"), 0, len(label.decode("ascii")),
+                                "label empty or too long")
```
Comment (eamanu) [inline Lib/encodings/idna.py]: an space is missing here, take care pep8

File: Lib/encodings/idna.py [hunk 2]
```diff
@@ -110,16 +114,18 @@ def ToUnicode(label):
...
         # Step 2: Perform nameprep
         label = nameprep(label)
         # It doesn't say this, but apparently, it should be ASCII now
         try:
             label = label.encode("ascii")
-        except UnicodeError:
-            raise UnicodeError("Invalid character in IDN label")
+        except UnicodeEncodeError:
+            raise UnicodeEncodeError("ascii", label.decode("ascii"), 0, len(label.decode("ascii")),
+                                    "Invalid character in IDN label")
```
Comment (eamanu) [inline Lib/encodings/idna.py]: same here

**Comment by srinivasreddy** [comment]
This is blatant copy of my PR https://github.com/python/cpython/pull/21165 . Why do you want to do it? @utkarsh261

**Comment by utkarsh261** [comment]
@srinivasreddy  These were the most logical changes, standard error messages, which I used from already existing code, what part of your "work" do you think i copied?
Was sending this PR to get familiar to the process mostly, if you think your work was copied, collaborators will notice that, please dont panic.

File: Lib/encodings/idna.py (hunk 1 above)
Comment (utkarsh261) [reply Lib/encodings/idna.py]: thanks!

**Comment by utkarsh261** [comment]
Closing this as @srinivasreddy already submitted a PR and I missed it earlier even though it looks like similar code.
Thanks.

**Pull request closed without merging**